Riot/Modules/Room/RoomViewController.m

```diff
@@ -6104,7 +6153,18 @@ - (void)mediaPickerCoordinatorBridgePresenter:(MediaPickerCoordinatorBridgePrese
     RoomInputToolbarView *roomInputToolbarView = [self inputToolbarViewAsRoomInputToolbarView];
     if (roomInputToolbarView)
     {
-        [roomInputToolbarView sendSelectedImage:imageData withMimeType:uti.mimeType andCompressionMode:BuildSettings.roomInputToolbarCompressionMode isPhotoLibraryAsset:YES];
+        // When the compression mode build setting hasn't been customised, use the media compression prompt setting to determine what to do.
```

Comment:
This code is duplicated3 times: we can create a `MediaCompressionHelper.swift` with `static getDefaultCompressionMode()` or `static var defaultCompressionMode`. Or update `RoomInputToolbarView` to handle the default compression mode logic inside.

Reply:
Yes that's much better, thanks.